frontend/packages/console-shared/src/hooks/useUserSettings.ts

```diff
@@ -0,0 +1,138 @@
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+
+// can't create project with name prefix with 'openshift-*', once we have proxy need to update
+const USER_SETTING_CONFIGMAP_NAMESPACE = 'console-user-settings';
+
+const getConfigResourceData = (userUid: string, key: string, defaultValue: string) => {
+  return [
+    {
+      kind: ConfigMapModel.kind,
+      namespace: USER_SETTING_CONFIGMAP_NAMESPACE,
+      isList: false,
+      name: `user-settings-${userUid}`,
+    },
+    {
+      apiVersion: ConfigMapModel.apiVersion,
+      kind: ConfigMapModel.kind,
+      metadata: {
+        name: `user-settings-${userUid}`,
+        namespace: USER_SETTING_CONFIGMAP_NAMESPACE,
+      },
+      data: {
+        [key]: defaultValue,
+      },
+    },
+  ];
+};
+
+// This won't be needed once we have proxy api
+const getProject = async () => {
+  try {
+    await k8sGet(ProjectModel, USER_SETTING_CONFIGMAP_NAMESPACE);
+  } catch {
+    await k8sCreate(ProjectRequestModel, {
+      metadata: {
+        name: USER_SETTING_CONFIGMAP_NAMESPACE,
+      },
+    });
+  }
+};
+
+const createConfigMap = async (configMapData: K8sResourceKind): Promise<boolean> => {
+  try {
+    await k8sCreate(ConfigMapModel, configMapData);
+    return true;
+  } catch (err) {
+    // eslint-disable-next-line no-console
+    console.error(err);
+    return false;
+  }
+};
+
+const updateConfigMap = async (configMap: ConfigMapKind, key: string, value: string) => {
+  if (value !== configMap.data?.[key]) {
+    const patch = [
+      {
+        op: 'replace',
+        path: `/data/${key}`,
+        value,
+      },
+    ];
+    try {
+      await k8sPatch(ConfigMapModel, configMap, patch);
+    } catch (err) {
+      // eslint-disable-next-line no-console
+      console.error(err);
+    }
+  }
+};
+
+const deseralizeData = (data: string) => {
+  try {
+    return JSON.parse(data);
+  } catch {
+    return data;
+  }
+};
+
+const seralizeData = <T>(data: T) => {
+  if (typeof data === 'string') {
+    return data;
+  }
+  return JSON.stringify(data);
+};
+
+export const useUserSettings = <T>(key: string, defaultValue?: T) => {
+  const userUid = useSelector(
+    (state: RootState) => state.UI.get('user')?.metadata?.uid ?? 'kubeadmin',
+  );
+  const [configMapResource, configMapData] = getConfigResourceData(
+    userUid,
+    key,
+    seralizeData(defaultValue ?? ''),
+  );
+  const [cfData, cfLoaded, cfLoadError] = useK8sWatchResource<K8sResourceKind>(configMapResource);
+  const [settings, setSettings] = React.useState();
+  const [loaded, setLoaded] = React.useState(false);
+
+  React.useEffect(() => {
+    if (cfLoadError || (!cfData && cfLoaded)) {
+      (async () => {
+        await getProject();
+        const cmCreated = await createConfigMap(configMapData);
+        if (!cmCreated) {
+          setLoaded(true);
+        }
+      })();
+    } else if (cfData && cfLoaded && (!settings || settings !== cfData.data?.[key])) {
```

Comment:
Settings should be allowed to be a falsey value.
You are comparing `settings` (a deserialized value) with `cfData.data[key]` which is a serialized value.

Reply:
missed here, doing the comparison with  serialized value now